x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/job_map.tsx

```diff
@@ -37,102 +37,78 @@ ${theme.euiColorLightShade}`,
-    <span>
-      {analyticsId
-        ? i18n.translate('xpack.ml.dataframe.analyticsMap.analyticsIdTitle', {
-            defaultMessage: 'Map for analytics ID {analyticsId}',
-            values: { analyticsId },
-          })
-        : i18n.translate('xpack.ml.dataframe.analyticsMap.modelIdTitle', {
-            defaultMessage: 'Map for trained model ID {modelId}',
-            values: { modelId },
-          })}
-    </span>
-  </EuiTitle>
-);
-
-interface GetDataObjectParameter {
-  id: string;
-  type: string;
-}
-
 interface Props {
   analyticsId?: string;
   modelId?: string;
 }
 
 export const JobMap: FC<Props> = ({ analyticsId, modelId }) => {
-  const [elements, setElements] = useState<cytoscape.ElementDefinition[]>([]);
-  const [nodeDetails, setNodeDetails] = useState({});
-  const [error, setError] = useState(undefined);
+  const [itemsDeleted, setItemsDeleted] = useState<boolean>(false);
+  const [resetCy, setResetCy] = useState<boolean>(false);
+  const {
+    elements,
+    error,
+    getDataWrapper,
+    isLoading,
+    message,
+    nodeDetails,
+    setElements,
+    setError,
+  } = useFetchAnalyticsMapData();
 
   const {
-    services: { notifications },
+    services: {
+      notifications,
+      application: { navigateToUrl },
+    },
   } = useMlKibana();
+  const urlGenerator = useMlUrlGenerator();
 
-  const getDataWrapper = async (params?: GetDataObjectParameter) => {
-    const { id, type } = params ?? {};
-    const treatAsRoot = id !== undefined;
-    let idToUse: string;
-
-    if (id !== undefined) {
-      idToUse = id;
-    } else if (modelId !== undefined) {
-      idToUse = modelId;
-    } else {
-      idToUse = analyticsId as string;
-    }
-
-    await getData(
-      idToUse,
-      treatAsRoot,
-      modelId !== undefined && treatAsRoot === false ? JOB_MAP_NODE_TYPES.TRAINED_MODEL : type
-    );
+  const redirectToAnalyticsManagementPage = async () => {
+    const url = await urlGenerator.createUrl({ page: ML_PAGES.DATA_FRAME_ANALYTICS_JOBS_MANAGE });
+    await navigateToUrl(url);
   };
 
-  const getData = async (idToUse: string, treatAsRoot: boolean, type?: string) => {
-    // Pass in treatAsRoot flag - endpoint will take job or index to grab jobs created from it
-    // TODO: update analyticsMap return type here
-    const analyticsMap: any = await ml.dataFrameAnalytics.getDataFrameAnalyticsMap(
-      idToUse,
-      treatAsRoot,
-      type
-    );
-
-    const { elements: nodeElements, details, error: fetchError } = analyticsMap;
-
-    if (fetchError !== null) {
-      setError(fetchError);
-    }
-
-    if (nodeElements && nodeElements.length === 0) {
-      notifications.toasts.add(
-        i18n.translate('xpack.ml.dataframe.analyticsMap.emptyResponseMessage', {
-          defaultMessage: 'No related analytics jobs found for {id}.',
-          values: { id: idToUse },
-        })
-      );
-    }
+  if (message !== undefined) {
+    notifications.toasts.add(message);
+  }
```

Comment:
I wonder if this could be problematic on unrelated rerenders, maybe we should put this inside a `useEffect()`?

Reply:
Good call! Moved into a useEffect in 626ab62bb02d68546b25cf3cbf527a0fd88cc1a9